public/app/features/trails/MetricSelectScene.tsx

```diff
@@ -15,20 +16,34 @@ import {
   SceneVariable,
   SceneCSSGridLayout,
   SceneCSSGridItem,
+  SceneObjectRef,
+  SceneQueryRunner,
+  VariableValueOption,
 } from '@grafana/scenes';
 import { VariableHide } from '@grafana/schema';
 import { Input, Text, useStyles2, InlineSwitch } from '@grafana/ui';
 
 import { getAutoQueriesForMetric } from './AutomaticMetricQueries/AutoQueryEngine';
 import { SelectMetricAction } from './SelectMetricAction';
 import { getVariablesWithMetricConstant, trailDS, VAR_FILTERS_EXPR, VAR_METRIC_NAMES } from './shared';
-import { getColorByIndex } from './utils';
+import { getColorByIndex, getTrailFor } from './utils';
+
+interface MetricPanel {
+  name: string;
+  index: number;
+  itemRef?: SceneObjectRef<SceneCSSGridItem>;
+  isHidden: boolean;
+  isPanel?: boolean;
+  loaded: boolean;
+}
 
 export interface MetricSelectSceneState extends SceneObjectState {
   body: SceneCSSGridLayout;
   showHeading?: boolean;
   searchQuery?: string;
   showPreviews?: boolean;
+  hideEmpty?: boolean;
+  previewMetricPanels: Record<string, MetricPanel>;
```

Comment:
think it might be better to move this from state to an instance property on MetricSelectScene instead, and name it something like previewCache. 

It does not need to be in the state because 
* You mutating it (ideally state should be immutable)
* A state update triggerd re-render and here the react Component does not actually care about previewMetricPanels , it's only used when building the layout.  
* So having it in state results in 24x re-renders of the MetricSelectScene react Component  (every hideMetric is called updating previewMetricPanels) 

If you handle this as an internal cache that optimizes how the layout is built I think it will work better, and there is no need to use SceneObjectRef.